nixos/modules/tasks/filesystems/overlayfs.nix

```diff
@@ -73,10 +76,22 @@ let
         description = ''
           The path to the workdir.
 
+          If the filesystem is `neededForBoot`, this will be prefixed with `/sysroot`,
+          unless `prefixWithSysroot` is set to `false`.
+
           This MUST be set if you set `upperdir`.
         '';
       };
 
+      prefixWithSysroot = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = ''
+          If enabled, `lowerdir`, `upperdir` and `workdir` will be prefixed with `/sysroot`.
```

Comment:
We should at least add a comment that this will still not make this work in the scripted initrd and maybe even an assertion that this cannot be used (as in it won't have the desired effect) in the scripted initrd.